### PR TITLE
fix: dragging floating ui components

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #8214 #7979

## Summary

- Update some component `open` handlers to call reposition before open and after open
  - Previously, they were just calling when open was set to true.
- Update `floating-ui` utility
  - Handle device pixel ratio: https://floating-ui.com/docs/misc#subpixel-and-accelerated-positioning 
  - Only set `top`, `left`, and `transform` when the component is open.
    - This makes it so that the positioned container is never going to "fly in" from anywhere. 
- Update sortable styles to hide any overflow on dragged elements.

### CODE REVIEW

**Comment by driskull** [comment]
@jcfranco this seems to fix the drag and drop issues with flying in from the right/left.

However, closed floating ui elements are still taking up space. What do you think?

**Comment by driskull** [comment]
Actually, if I add overflow hidden on elements that are dragged it seems to be good.

Can you review/test? @jcfranco

File: packages/calcite-components/src/utils/floating-ui.ts [hunk 1]
```diff
@@ -145,15 +150,15 @@ export const positionFloatingUI =
...
+    const { open } = component;
 
     Object.assign(floatingEl.style, {
       visibility,
       pointerEvents,
       position,
-      top: "0",
-      left: "0",
-      transform,
+      transform: open ? `translate(${roundByDPR(x)}px,${roundByDPR(y)}px)` : null,
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/floating-ui.ts:159]: Type-wise, should we pass `""` vs `null`?

**Review by jcfranco — APPROVED** [review]
🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈
🎈🆙🆙🆙🆙🎈🆙🎈🎈🎈🎈🎈🆙🎈🆙🆙🆙🆙🎈🎈🆙🆙🆙🎈🎈🆙🆙🎈🎈🆙🎈🎈🎈🆙🎈🆙🆙🆙🆙🎈🆙🎈
🎈🆙🎈🎈🆙🎈🆙🎈🎈🎈🎈🎈🆙🎈🆙🎈🎈🎈🎈🆙🎈🎈🎈🎈🆙🎈🎈🆙🎈🆙🆙🎈🆙🆙🎈🆙🎈🎈🎈🎈🆙🎈
🎈🆙🆙🆙🆙🎈🆙🎈🎈🆙🎈🎈🆙🎈🆙🆙🆙🎈🎈🎈🆙🆙🎈🎈🆙🎈🎈🆙🎈🆙🎈🆙🎈🆙🎈🆙🆙🆙🎈🎈🆙🎈
🎈🆙🎈🎈🆙🎈🆙🎈🆙🎈🆙🎈🆙🎈🆙🎈🎈🎈🎈🎈🎈🎈🆙🎈🆙🎈🎈🆙🎈🆙🎈🎈🎈🆙🎈🆙🎈🎈🎈🎈🎈🎈
🎈🆙🎈🎈🆙🎈🎈🆙🎈🎈🎈🆙🎈🎈🆙🆙🆙🆙🎈🆙🆙🆙🎈🎈🎈🆙🆙🎈🎈🆙🎈🎈🎈🆙🎈🆙🆙🆙🆙🎈🆙🎈
🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈🎈

**Comment by jcfranco** [comment]
> Can you review/test?

FYI, I only reviewed. Haven't had a chance to test.

File: packages/calcite-components/src/utils/floating-ui.ts (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/utils/floating-ui.ts]: Updated.

**Comment by driskull** [comment]
@jcfranco good to merge?

**Pull request merged**